graph/src/data/store/scalar.rs

```diff
@@ -10,14 +13,158 @@ use stable_hash::{
 };
 use std::convert::{TryFrom, TryInto};
 use std::fmt::{self, Display, Formatter};
+use std::io::Write;
 use std::ops::{Add, Div, Mul, Rem, Sub};
 use std::str::FromStr;
 
 pub use num_bigint::Sign as BigIntSign;
 
-// Caveat: The exponent is currently an i64 and may overflow.
-// See https://github.com/akubera/bigdecimal-rs/issues/54.
-pub type BigDecimal = bigdecimal::BigDecimal;
+/// All operations on `BigDecimal` return a normalized value.
+// Caveat: The exponent is currently an i64 and may overflow. See
+// https://github.com/akubera/bigdecimal-rs/issues/54.
+// Using `#[serde(from = "BigDecimal"]` makes sure deserialization calls `BigDecimal::new()`.
+#[derive(
+    Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, AsExpression, FromSqlRow,
+)]
+#[serde(from = "BigDecimal")]
+#[sql_type = "diesel::sql_types::Numeric"]
```

Comment:
I would prefer not to add any more diesel-dependent code to the `graph` crate. Could this be done in the diesel store? (Might require an extra newtype around `BigDecimal`?)

Reply:
It would require a newtype, and between this and a newtype, I see this as the lesser evil.